onadata/apps/api/tests/viewsets/test_attachment_viewset.py

```diff
@@ -188,6 +188,72 @@ def test_list_view(self):
         self.assertTrue(isinstance(response.data, list))
         self.assertEqual(len(response.data), 0)
 
+    def test_list_view_surfaces_current_attachment(self):
+        self._submit_transport_instance_w_attachment()
+        filename = "1335783522564.JPG"
+        path = os.path.join(self.main_directory, 'fixtures', 'transportation',
+                            'instances', self.surveys[0], filename)
+        media_file = django_file(path, 'video2', 'image/jpeg')
+
+        Attachment.objects.create(
+            instance=self.xform.instances.first(),
+            mimetype='video/mp4',
+            extension='MP4',
+            name=filename,
+            media_file=media_file)
+
+        Attachment.objects.create(
+            instance=self.xform.instances.first(),
+            mimetype='application/pdf',
+            extension='PDF',
+            name=filename,
+            media_file=media_file)
+        Attachment.objects.create(
+            instance=self.xform.instances.first(),
+            mimetype='text/plain',
+            extension='TXT',
+            name=filename,
+            media_file=media_file)
+        Attachment.objects.create(
+            instance=self.xform.instances.first(),
+            mimetype='audio/mp3',
+            extension='MP3',
+            name=filename,
+            media_file=media_file)
+
+        request = self.factory.get('/', **self.extra)
+        response = self.list_view(request)
+        self.assertNotEqual(response.get('Cache-Control'), None)
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(isinstance(response.data, list))
+        self.assertEqual(len(response.data), 5)
+
+        # test when the attachment is soft deleted
+        self.attachment.soft_delete(user=self.user)
+        self.attachment.save()
+
+        request = self.factory.get('/', **self.extra)
+        response = self.list_view(request)
+
+        self.assertEqual(len(response.data), 4)
+
+    def test_soft_delete_action_returns_correct_user(self):
+        self._submit_transport_instance_w_attachment()
+
+        request = self.factory.get('/', **self.extra)
+        response = self.list_view(request)
+        self.assertEqual(len(response.data), 1)
+
+        # test when the attachment is soft deleted
+        self.attachment.soft_delete(user=self.user)
+        self.attachment.save()
```

Comment:
Why do we call `save()` here ?

Reply:
I see how this may be repetitive. Removing that.